content/blog/an-example-application-for-processing-stock-market-trade-data-on-the-mapr-converged-data-platform.md

```diff
@@ -0,0 +1,291 @@
+
+### Step 5: Persist stream data in a database
+
+We'll explain two ways in which streaming data can be persisted into long-term storage. First, we'll see how this can be done with MapR-DB, then we'll see how this can be done with Apache Hive.  
+
+#### Persist stream data with MapR-DB
+
+The `Persister.java` class uses Spark SQL to persist JSON records from MapR Streams into MapR-DB.
+
+This class can be run with the following command:
+
+```
+java -cp `mapr classpath`:/home/mapr/finserv-application-blueprint/target/nyse-taq-streaming-1.0.jar com.mapr.demo.finserv.Persister -topics /user/mapr/taq:sender_0310,/user/mapr/taq:sender_0410 -table /user/mapr/ticktable -droptable -verbose
+```
+
+This creates a stream consumer that persists trades from senders #0310 and #0410 to MapR-DB in a table located at /mapr/my.cluster.com/user/mapr/ticktable (which will be overwritten if it already exists, per the `-droptable` option). That command will only see *new* messages in the trades topic because it tails the log, so run the following command to put more trade data into the stream:
+
+```
+java -cp `mapr classpath`:/home/mapr/nyse-taq-streaming-1.0.jar:/home/mapr/finserv-application-blueprint/src/test/resources com.mapr.demo.finserv.Run consumer /user/mapr/taq:trades 3 &
+java -cp `mapr classpath`:/home/mapr/nyse-taq-streaming-1.0.jar com.mapr.demo.finserv.Run producer /home/mapr/finserv-application-blueprint/data/ /user/mapr/taq:trades
+```
+
+Here are some examples of how you can query the table created by the Persister:
+
+Query the MapR-DB table with dbshell:
+
+```
+mapr dbshell
+  maprdb mapr:> find /user/mapr/ticktable
+```
+
+Query the MapR-DB table from the Apache Drill command line:
+
+```
+/opt/mapr/drill/drill-*/bin/sqlline -u jdbc:drill:
+0: jdbc:drill:> SELECT * FROM dfs.`/user/mapr/ticktable` LIMIT 10;
+```
+
+Query the MapR-DB table from the Apache Drill web interface, as shown below:
+
+<img src = "https://github.com/mapr-demos/finserv-application-blueprint/raw/master/images/drill_query.png" width=600px>
+
+#### Persist Stream Data with Apache Hive
+
+The `SparkStreamingToHive` class uses the Spark Streaming API to copy messages from the tail of streaming topics to Hive tables that can be analyzed in Zeppelin. Zeppelin can't directly access stream topics, so we use this utility to access streaming data from Zeppelin. Here's how to run this class:
+
+```
+/opt/mapr/spark/spark-2.0.1/bin/spark-submit --class com.mapr.demo.finserv.SparkStreamingToHive /home/mapr/nyse-taq-streaming-1.0-jar-with-dependencies.jar --topics <topic1>,<topic2>... --table <destination Hive table>
+```
+
+That command will only see *new* messages in the taq:trades topic because it tails the log, so when it says "Waiting for messages" then run the following command to put more trade data into the stream. This command was described in [step 4](https://github.com/mapr-demos/finserv-application-blueprint#step-4-run-the-producer):
```

Comment:
Not sure you actually need to hotlink "step4" here. Step4 is described earlier in the blog. 
This link directs the readers to a Github repo https://github.com/mapr-demos/finserv-application-blueprint#step-4-run-the-producer 
and the blog here seems to be the "echo" blog of this GitHub repo.